src/Components/CreateImageWizardV2/steps/TargetEnvironment/Azure/AzureSourcesSelect.tsx

```diff
@@ -31,51 +39,45 @@ const AzureSourcesSelect = ({ label, isRequired, className, ...props }) => {
     isError,
     refetch,
   } = useGetSourceListQuery({ provider: 'azure' });
-  const sources = extractProvisioningList(rawSources);
 
   const {
     data: sourceDetails,
     isFetching: isFetchingDetails,
     isSuccess: isSuccessDetails,
     isError: isErrorDetails,
   } = useGetSourceUploadInfoQuery(
-    { id: selectedSourceId },
+    // @ts-ignore
```

Comment:
We shouldn't be using @ts-ignore without a very good reason. In this case, there is a type difference between `id` (which must be a number) and `azureSource` (which is string or undefined).

We should expect azureSource to be castable to number... what if we wrote a helper function, something like `convertAzureSourceIdToNumber()`, that attempts to cast azureSource to Number in a try...catch clause and throws a type error if it fails?